### PR TITLE
Fix a bug where running jobs are not getting picked up when server restarts (or when cluster reshards)

### PR DESCRIPTION
**Pull request by derekchiang**

### CODE REVIEW

File: src/server/pps/hash.go [hunk 1]
```diff
@@ -20,10 +20,10 @@ func NewHasher(jobModulus uint64, pipelineModulus uint64) *Hasher {
...
 // HashJob computes and returns the hash of a job.
 func (s *Hasher) HashJob(jobID string) uint64 {
-	return uint64(adler32.Checksum([]byte(jobID))) % s.PipelineModulus
+	return uint64(adler32.Checksum([]byte(jobID))) % s.JobModulus
 }
 
 // HashPipeline computes and returns the hash of a pipeline.
 func (s *Hasher) HashPipeline(pipelineName string) uint64 {
-	return uint64(adler32.Checksum([]byte(pipelineName))) % s.JobModulus
+	return uint64(adler32.Checksum([]byte(pipelineName))) % s.PipelineModulus
```
Comment (jdoliner) [inline src/server/pps/hash.go]: ap_090911089838_sq-3271237f28995f6530d9634ff27228cae88e3440-s900-c85

**Comment by jdoliner** [comment]
All LGTM, is there already a test which reproduces this already? If not would you mind adding one?

File: src/server/pps/hash.go (hunk 1 above)
Comment (derekchiang) [inline src/server/pps/hash.go]: Yeah that was pretty hilarious

**Comment by jdoliner** [comment]
LGTM

**Pull request merged**